### ISSUE TITLE
Support:SSR

### ISSUE THREAD

**Issue by gHashTag** [reframejs/reframe#28]
At first glance interested in your project and most of all I'm worried about the question of creating a universal application. I tried to build my webpack config, but it's a very long story. And many searches have brought you.
How to configure routing so that after `reframe build` and `reframe server` it does not show an error 404?
![ssr mov](https://user-images.githubusercontent.com/6774813/39643752-770c93c6-4fde-11e8-9bd4-5a178e4fe377.gif)

**Comment by gHashTag** [comment]
It's very likely that your object is what I've been looking for so long =)
Before I start migrating with CRA, could give recommendations on your comment
> Just out of curiosity: Why are you using react-router's <Link> and <Route> instead of having different page config for each page /, /about and /topics?

I did not quite understand what I was doing wrong.
[Source code here](https://github.com/gHashTag/reframe)

**Comment by brillout** [comment]
Just looked at your last commit (https://github.com/gHashTag/reframe/commit/b123d62d99c17eb9b7db69ac50abb343acd13046)

You changed the route, SSR should be working for you now.

As for my question: What you are doing is totally fine. It's just that what you are doing is in essence an SPA: You have exactly *one* page config hence your entire app will be bundled into that *one* page.

Alternatively you can defined several page configs:

```js
// pages/home.config.js
const HomePage = {
  route: '/',
  view: homeView,
};
export default HomePage;
```

```js
// pages/contacts.config.js
const ContactsPage = {
  route: '/contacts',
  view: contactsView,
};
export default ContactsPage;
```

Basically: One page config file saved under your `pages/` directory per page.

What you are doing is totally fine. But spliting your app into several page configs is more performant: When going to `/contacts` the user will load only the contacts page's JavaScript instead of loading the entire app.
Does that make sense?

> It's very likely that your project is what I've been looking for so long =)

Cool =) Feel free to hit me up on Discord and open further GitHub issues ;-)
